### ISSUE TITLE
Adding pos/rot key from 2D freezes the editor (regression)

### ISSUE THREAD

**Issue by reduz** [godotengine/godot#18614]
![image](https://user-images.githubusercontent.com/6265307/39632688-2e7dbb14-4f8c-11e8-90af-2d856c798a4e.png)

Demo Scene
[s2d.zip](https://github.com/godotengine/godot/files/1974732/s2d.zip)

Please use HEAD to test

**Comment by eon-s** [comment]
Got that yesterday on 3.0.2, touching the animation menu also fades the editor like trying to show a popup but never shown, the editor remains usable but in faded color state.

**Comment by Vega-KH** [comment]
Testing on Win10 (HEAD), it seems the issue doesn't happen if selected animation in the Animationplayer already has at least one track with at least one key already defined.

Steps to test:

1. Add an animation to the AnimationPlayer using the bottom "Animation" plugin.
2. Select something that can be animated in the scene, then click the KEY next to the property in the inspector.
3. Now use the KEY button to create another keyframe using the top toolbar. Notice no error.

So basically, it only happens when you use the top toolbar to create the first track in an animation.

**Comment by noidexe** [comment]
Still present in 3.0.3 rc2

**Comment by adrianperreault** [comment]
![gif](https://www.dropbox.com/s/ljq2pe5ur189q6f/godot-error.gif?raw=1)

Same issue here with the dimming like it's trying to show a modal.
(That's upon clicking the key icon)

The console output is displaying the following:

```
adrian@adrianscomp ~> ERROR: get_viewport_rect: Condition ' !is_inside_tree() ' is true. returned: Rect2()
   At: scene/2d/canvas_item.cpp:864.
ERROR: show_modal: Condition ' !is_inside_tree() ' is true.
   At: scene/gui/control.cpp:2121.
ERROR: get_global_transform: Condition ' !is_inside_tree() ' is true. returned: get_transform()
   At: scene/2d/canvas_item.cpp:379.
ERROR: get_viewport_rect: Condition ' !is_inside_tree() ' is true. returned: Rect2()
   At: scene/2d/canvas_item.cpp:864.
ERROR: grab_focus: Condition ' !is_inside_tree() ' is true.
   At: scene/gui/control.cpp:2095.
ERROR: get_tree: Condition ' !data.tree ' is true. returned: __null
   At: ./scene/main/node.h:263.
ERROR: texture_set_size_override: Condition ' p_width <= 0 || p_width > 16384 ' is true.
   At: drivers/gles3/rasterizer_storage_gles3.cpp:1093.
ERROR: texture_set_size_override: Condition ' p_width <= 0 || p_width > 16384 ' is true.
   At: drivers/gles3/rasterizer_storage_gles3.cpp:1093.
ERROR: get_pixel: Index p_x=288 out of size (width=64)
   At: core/image.cpp:1950.
ERROR: get_pixel: Index p_x=281 out of size (width=64)
   At: core/image.cpp:1950.
ERROR: get_pixel: Index p_x=113 out of size (width=64)
   At: core/image.cpp:1950.
ERROR: get_pixel: Index p_x=84 out of size (width=64)
   At: core/image.cpp:1950.
ERROR: get_pixel: Index p_y=95 out of size (height=64)
   At: core/image.cpp:1951.
ERROR: get_pixel: Index p_y=98 out of size (height=64)
   At: core/image.cpp:1951.
ERROR: get_pixel: Index p_y=798 out of size (height=64)
   At: core/image.cpp:1951.
ERROR: get_pixel: Index p_x=302 out of size (width=64)
   At: core/image.cpp:1950.
ERROR: get_pixel: Index p_x=302 out of size (width=64)
   At: core/image.cpp:1950.
ERROR: get_pixel: Index p_x=313 out of size (width=64)
   At: core/image.cpp:1950.
ERROR: get_pixel: Index p_x=354 out of size (width=64)
   At: core/image.cpp:1950.
ERROR: get_pixel: Index p_x=398 out of size (width=64)
   At: core/image.cpp:1950.
ERROR: get_pixel: Index p_x=446 out of size (width=64)
   At: core/image.cpp:1950.
ERROR: get_pixel: Index p_x=451 out of size (width=64)
   At: core/image.cpp:1950.
ERROR: show_modal: Condition ' data.MI != __null ' is true.
   At: scene/gui/control.cpp:2127.
ERROR: show_modal: Condition ' data.MI != __null ' is true.
   At: scene/gui/control.cpp:2127.
ERROR: show_modal: Condition ' data.MI != __null ' is true.
   At: scene/gui/control.cpp:2127.
ERROR: get_pixel: Index p_x=529 out of size (width=64)
   At: core/image.cpp:1950.
ERROR: get_pixel: Index p_x=787 out of size (width=64)
   At: core/image.cpp:1950.
ERROR: get_pixel: Index p_x=979 out of size (width=64)
   At: core/image.cpp:1950.
ERROR: show_modal: Condition ' data.MI != __null ' is true.
   At: scene/gui/control.cpp:2127.
ERROR: show_modal: Condition ' data.MI != __null ' is true.
   At: scene/gui/control.cpp:2127.
ERROR: show_modal: Condition ' data.MI != __null ' is true.
   At: scene/gui/control.cpp:2127.
ERROR: get_pixel: Index p_x=934 out of size (width=64)
   At: core/image.cpp:1950.
ERROR: get_pixel: Index p_x=707 out of size (width=64)
   At: core/image.cpp:1950.
ERROR: get_pixel: Index p_x=378 out of size (width=64)
   At: core/image.cpp:1950.
ERROR: get_pixel: Index p_x=295 out of size (width=64)
   At: core/image.cpp:1950.
ERROR: get_pixel: Index p_x=238 out of size (width=64)
   At: core/image.cpp:1950.
```

**Comment by noidexe** [comment]
@akien-mga This is still happening with 3.0.3 rc 3. Being that it is a regression and it probably affects the workflow of a big chunk of the user base, shouldn't it be closed before marking a build as RC?

**Comment by akien-mga** [comment]
CC @hpvb

**Comment by akien-mga** [comment]
> Got that yesterday on 3.0.2, touching the animation menu also fades the editor like trying to show a popup but never shown, the editor remains usable but in faded color state.

Can't reproduce on 3.0.2-stable. I can reproduce it in 3.0.3-rc3 though, the "Insert new animation" dialog doesn't show up properly, but still prevents any input.

**Comment by akien-mga** [comment]
I'm bisecting to find out what cherry-pick caused the regression in the 3.0 branch.

**Comment by akien-mga** [comment]
The regression comes from #18172 (d42b176), cherry-picked in the 3.0 branch as 826108e.
